projects/igniteui-angular/src/lib/core/styles/components/button/_button-theme.scss

```diff
@@ -135,6 +140,14 @@
             $focus-foreground: text-contrast($focus-background);
         }
 
+        @if not($focus-hover-background) and $background {
+            $focus-hover-background: color.scale($background, $lightness: 5%);
```

Comment:
This will not work if the input for background is a reference to a CSS variable. You should check if `$background` is of type color before doing any color operations on it.